python/dask_cudf/dask_cudf/_expr/__init__.py

```diff
@@ -1 +1,14 @@
-# Copyright (c) 2024, NVIDIA CORPORATION.
+# Copyright (c) 2024-2025, NVIDIA CORPORATION.
+
+try:
+    from dask.dataframe import dask_expr  # noqa: F401
+
+except ImportError:
+    # TODO: Remove when pinned to dask>2024.12.1
+    import dask.dataframe as dd
+
+    if not dd._dask_expr_enabled():
+        raise ValueError(
+            "The legacy DataFrame API is not supported in dask_cudf>24.12. "
+            "Please enable query-planning, or downgrade to dask_cudf<=24.12"
```

Comment:
I think instructing to downgrade doesn't make sense, we currently pin exactly [`2024.12.1`](https://github.com/rapidsai/rapids-dask-dependency/blob/29af6cced43258ed36fac42ffb4faf41a497ae1d/pyproject.toml#L15-L16) and will like later pin `>=2024.12.1` or something newer than that. Should this be removed?

Reply:
Sorry, my bad, I misread and thought this referred to Dask version. In any case, I'm not sure if we should encourage users to downgrade to an older RAPIDS version. Maybe we should at least clarify downgrading is discouraged but is a short-term alternative for them if absolutely necessary, WDYT?